lib/tx_build/resource.ex

```diff
@@ -1,4 +1,4 @@
-defmodule Stellar.TxBuild.Spec do
+defmodule Stellar.TxBuild.Resource do
```

Comment:
IMO, `Resource` is a confusing name for this specification because that word is never used in the Stellar world and it also means several different things in the computing world. Since we are "composing," words like `Component` or `Part` are more appropriate. Another idea is to see what the specification tries to describe, in this case, looks like it is trying to describe `XDRAble` things.  What do you think?

Reply:
🤔  you're right, probably using the word **resource** is not a good name for this spec since "Resource" is the way the Horizon API endpoints are organized. 

These structures in `tx_build` have the responsibility of taking input-data and converting it into XDR constructions, we can use something like  `XDR`, `XDRBuild`, `XDRConstruction` or move this into an `xdr` namespace in the `tx_build` abstraction and call it `XDR.Spec` what do you think @luishurtado?

Reply:
I don't see it is required a new layer of abstraction here. The name `XDR` will do it.